Code/Framework/AzCore/AzCore/Component/ComponentApplicationLifecycle.cpp

```diff
@@ -56,15 +56,21 @@ namespace AZ::ComponentApplicationLifecycle
     }
 
     bool RegisterHandler(AZ::SettingsRegistryInterface& settingsRegistry, AZ::SettingsRegistryInterface::NotifyEventHandler& handler,
-        AZ::SettingsRegistryInterface::NotifyCallback callback, AZStd::string_view eventName)
+        AZ::SettingsRegistryInterface::NotifyCallback callback, AZStd::string_view eventName, bool autoRegisterEvent)
     {
         using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
         using Type = AZ::SettingsRegistryInterface::Type;
         using NotifyEventHandler = AZ::SettingsRegistryInterface::NotifyEventHandler;
 
-        if (!ValidateEvent(settingsRegistry, eventName))
+        // Some systems may attempt to register a handler before the settings registry has been loaded
+        // If so, this flag lets them automatically register an event if it hasn't yet been registered.
+        // RegisterEvent calls validate event.
+        if ((!autoRegisterEvent && !ValidateEvent(settingsRegistry, eventName)) ||
+            (autoRegisterEvent && !RegisterEvent(settingsRegistry, eventName)))
```

Comment:
This can be simplified to reduce one of the `autoRegisterEvent` calls
```
if (!ValidateEvent(settingsRegistry, eventName) && (!autoRegisterEvent || !RegisterEvent(settingsRegistry, eventName)))
```

Reply:
autoRegisterEvent is a bool, I switched it to what I have from what you're suggesting because your version calls ValidateEvent twice, the first thing called in RegisterEvent is ValidateEvent.